src/main/java/org/fcrepo/client/FcrepoLink.java

```diff
@@ -26,23 +30,43 @@
  */
 public class FcrepoLink {
 
+    // Pattern to match parameters in a link header, such as rel="acl" or type=unquoted, separated by ';'
+    private static Pattern LINK_PARAM_PATTERN = Pattern.compile("(\\w+) *= *(\"[^\"]+\"|[^\",;]+)\\s*(;|$)");
```

Comment:
Do we have some more transparent way to parse Link headers? This regex seems like a maintenance headache. Did you inspect the implementation for: https://javaee.github.io/javaee-spec/javadocs/javax/ws/rs/core/Link.Builder.html ?

Reply:
This regex isn't doing all that much, it matches `name="value"` pairs, with out without the quotes, with ; separators between them. I did look at the jersey implementation of the Link classes a fair amount, although i didn't for this part before now. Their equivalent is:
https://github.com/jersey/jersey/blob/master/core-common/src/main/java/org/glassfish/jersey/message/internal/LinkProvider.java#L105-L117
We could use the same approach if you prefer. I guess i don't personally find it easier to read, but I may rely on regexes more than necessary.

Reply:
Regex's are handy and helpful. I have maintenance concerns when I look at a regex and it is not immediately obvious what it is capturing.
There is the famous (oft-misrepresented) [quote](https://groups.google.com/d/msg/alt.religion.emacs/DR057Srw5-c/n1WCMEw5iCkJ):
> Some people, when confronted with a problem, think “I know, I'll use regular expressions.”   Now they have two problems.

It is a judgement call, and I trust your judgement... but wanted to raise the flag of concern.